explorer/e2e/testFunctions.ts

```diff
@@ -522,4 +541,255 @@ export async function testClearAll(
+    .first()
+    .getByRole("cell")
+    .first()
+    .getByRole("link");
+
+// Backpages tests
+export async function testExportBackpage(
+  context: BrowserContext,
+  page: Page,
+  tab: TabDescription
+): Promise<void> {
+  if (tab.backpageExportButtons == null || tab.backpageAccessTags == null) {
+    // Fail if this test is ran on a tab without defined backpages
+    await expect(false);
+    return;
+  }
+  // Goto the specified tab
+  await page.goto(tab.url);
+  // Expect to find row with a granted status indicator
+  const grantedRowLocator = getRowLocatorByAccess(
+    page,
+    tab.backpageAccessTags.grantedShortName
+  );
+  await expect(grantedRowLocator).toBeVisible();
+  // Click into the selected row
+  await grantedRowLocator.dispatchEvent("click");
+  await expect(
+    page.getByText(tab.backpageExportButtons.detailsName)
+  ).toBeVisible();
+  // Click the "Export" tab
+  await page
+    .getByText(tab.backpageExportButtons.exportTabName, {
+      exact: true,
+    })
+    .click();
+  await expect(page).toHaveURL(tab.backpageExportButtons.exportUrlRegExp);
+  await expect(page.getByRole("checkbox").first()).toBeVisible();
+  const firstButtonLocator = page.getByRole("button", {
+    name: tab.backpageExportButtons.firstButtonName,
+  });
+  await expect(firstButtonLocator).toBeEnabled();
+  // Select all checkboxes on the pages
+  const checkboxLocators = await page.getByRole("checkbox").all();
+  for (const checkboxLocator of checkboxLocators) {
+    if (!(await checkboxLocator.isChecked())) {
+      await checkboxLocator.click();
+      await expect(checkboxLocator).toBeChecked();
+      await expect(checkboxLocator).toBeEnabled({ timeout: 10000 });
+    }
+  }
+  await expect(firstButtonLocator).toBeEnabled({ timeout: 10000 });
+  // Uncheck all checkboxes except one in each table, to reduce overhead
+  for (const tableLocator of await page.getByRole("table").all()) {
+    const checkboxLocatorsInTable = await tableLocator
+      .getByRole("checkbox")
+      .all();
+    for (const checkboxLocator of checkboxLocatorsInTable.slice(2)) {
+      await checkboxLocator.click();
+      await expect(checkboxLocator).not.toBeChecked();
+      await expect(checkboxLocator).toBeEnabled({ timeout: 10000 });
+    }
+  }
+  // Click the "Request Link" button
+  await expect(firstButtonLocator).toBeEnabled({ timeout: 10000 });
+  await firstButtonLocator.click();
+  await expect(
+    page.getByText(tab.backpageExportButtons.firstLoadingMessage, {
+      exact: true,
+    })
+  ).toBeVisible();
+  await expect(
+    page.getByText(tab.backpageExportButtons.secondLandingMessage, {
+      exact: true,
+    })
+  ).toBeVisible({ timeout: 60000 });
+  const secondButtonLocator = page.getByRole("button", {
+    name: tab.backpageExportButtons?.secondButtonName,
+  });
+  await expect(secondButtonLocator).toBeEnabled();
+  // Click the "Open Terra" Button and await a new browser tab
```

Comment:
Interesting! Apologies if you've mentioned this before but we'll need a way to generalize this for other exports (e.g. bulk download) where the export action is not opening a new tab. I think for now let's just generalize this comment and we can modify the functionality at a later point when we add tests for other exports.
```suggestion
  // Click the export action button and await a new browser tab
```

Reply:
Makes sense! When we add tests for other exports I can have this function take a second function as an argument to deal with the result of the export action button.